tests/System.Text.Primitives.Tests/Encoding/RandomTests.cs

```diff
@@ -147,8 +147,8 @@ public void ToString(string expected, Utf8String s)
         [Theory]
         [MemberData("StringEqualsTestCases_EmptyStrings")]
         [MemberData("StringEqualsTestCases_SimpleStrings")]
-        public unsafe void StringEquals(bool expected, Utf8String s1, Utf8String s2)
-        {
+        public unsafe void StringEqualsWithThreeArgs(bool expected, Utf8String s1, Utf8String s2)
+        {   // TODO: investigate why the tests fail if we have two methods with the same name StringEquals
```

Comment:
What do you mean by that? Two methods with the same name on different types? Are the types tests?